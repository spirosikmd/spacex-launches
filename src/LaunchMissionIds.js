import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

const LaunchMissionIds = ({ missionIds }) => {
  return (
    <Typography color="textSecondary" component="p">
      ID{missionIds.length > 1 ? 's' : null}: {missionIds.join(', ')}
    </Typography>
  );
};

LaunchMissionIds.propTypes = {
  missionIds: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default React.memo(LaunchMissionIds);
