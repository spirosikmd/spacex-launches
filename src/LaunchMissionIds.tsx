import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

interface LaunchMissionIdsProps {
  missionIds: string[];
}

const LaunchMissionIds = ({ missionIds }: LaunchMissionIdsProps) => {
  return (
    <Typography color="textSecondary" component="p">
      ID{missionIds.length > 1 ? 's' : null}: {missionIds.join(', ')}
    </Typography>
  );
};

LaunchMissionIds.propTypes = {
  missionIds: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default React.memo(LaunchMissionIds);
