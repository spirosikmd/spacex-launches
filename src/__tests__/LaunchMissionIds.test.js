import LaunchMissionIds from '../LaunchMissionIds';

describe('LaunchMissionIds', () => {
  let props;

  beforeEach(() => {
    props = {
      missionIds: ['id1'],
    };
  });

  it('renders', () => {
    expect(mountComponent(LaunchMissionIds, props)).toMatchSnapshot();
  });

  it('render IDs with multiple ids', () => {
    props.missionIds = ['id1', 'id2'];
    expect(mountComponent(LaunchMissionIds, props)).toMatchSnapshot();
  });
});
