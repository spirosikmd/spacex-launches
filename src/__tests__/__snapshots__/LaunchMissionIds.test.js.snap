// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchMissionIds render IDs with multiple ids 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    IDs: id1, id2
  </p>
</DocumentFragment>
`;

exports[`LaunchMissionIds renders 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    ID: id1
  </p>
</DocumentFragment>
`;
